src/pivot-table/data/use-pv-data.ts

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useMemo, useState } from "@nebula.js/stardust";
+import type { PageInfo } from "../../hooks/use-pivot-table";
+import type {
+  Data,
+  HeadersData,
+  LayoutService,
+  LeftDimensionData,
+  MeasureData,
+  TopDimensionData,
+} from "../../types/types";
+import createHeadersData from "./headers-data";
+import { addPageToLeftDimensionData, createLeftDimensionData } from "./left-dimension-data";
+import { addPageToMeasureData, createMeasureData } from "./measure-data";
+import { addPageToTopDimensionData, createTopDimensionData } from "./top-dimension-data";
+
+interface UsePVDataProps {
+  qPivotDataPages: EngineAPI.INxPivotPage[];
+  layoutService: LayoutService;
+  pageInfo: PageInfo;
+}
+
+export const usePVData = ({ qPivotDataPages, layoutService, pageInfo }: UsePVDataProps): Data => {
```

Comment:
Is this hook really needed? Can you not use `useLoadDataPages` to get the "initial" data for a given page? That hook just need to know the page that is currently active to load data